src/explorer.ts

```diff
@@ -32,7 +35,7 @@ export interface KubernetesObject {
     readonly id: string;
     readonly metadata?: any;
     getChildren(kubectl: Kubectl, host: Host): vscode.ProviderResult<KubernetesObject[]>;
-    getTreeItem(): vscode.TreeItem | Thenable<vscode.TreeItem>;
+    getTreeItem(type: string | undefined): vscode.TreeItem | Thenable<vscode.TreeItem>;
```

Comment:
I'm not keen on this - the only place the `type` information can come from is surely the object itself, so why would it need to pass it to its own member function?  Suggest that things that are interested in `type` implement it as a field and inspect it from their `getTreeItem` implementations...?